storage/replica_command.go

```diff
@@ -398,13 +414,29 @@ func (r *Replica) InternalRangeLookup(batch engine.Engine, args proto.InternalRa
 		rangeCount = 1
 	}
 
-	// We want to search for the metadata key just greater than args.Key. Scan
-	// for both the requested key and the keys immediately afterwards, up to
-	// MaxRanges.
-	startKey, endKey := keys.MetaScanBounds(args.Key)
-	// Scan for descriptors.
-	kvs, intents, err := engine.MVCCScan(batch, startKey, endKey, rangeCount,
-		args.Timestamp, consistent, args.Txn)
+	var kvs []proto.KeyValue
+	var intents []proto.Intent
+	var err error
+	if !args.Reverse {
+		// We want to search for the metadata key just greater than args.Key. Scan
+		// for both the requested key and the keys immediately afterwards, up to
+		// MaxRanges.
+		startKey, endKey := keys.MetaScanBounds(args.Key)
+		// Scan for descriptors.
+		kvs, intents, err = engine.MVCCScan(batch, startKey, endKey, rangeCount,
+			args.Timestamp, consistent, args.Txn)
+	} else {
+		// We want to search for the metadata key just less than args.Key. Reverse
```

Comment:
currently, it's not "just less": you're searching "less or equal" (but see my suggestion above).